code/modules/power/supermatter/supermatter.dm

```diff
@@ -384,7 +504,106 @@
 	icon_state = "darkmatter"
 	anchored = 1
 	gasefficency = 0.15
-	explosion_power = 20
+	explosion_power = 35
+
+/obj/machinery/power/supermatter_shard/proc/supermatter_pull(turf/center, pull_range = 10)
+	playsound(src.loc, 'sound/weapons/marauder.ogg', 100, 1, extrarange = 7)
+	for(var/atom/P in orange(pull_range,center))
+		if(istype(P, /atom/movable))
+			var/atom/movable/pulled_object = P
+			if(ishuman(P))
+				var/mob/living/carbon/human/H = P
+				H.apply_effect(2, WEAKEN, 0)
+			if(pulled_object && !pulled_object.anchored && !ishuman(P))
+				step_towards(pulled_object,center)
+				step_towards(pulled_object,center)
+				step_towards(pulled_object,center)
+				step_towards(pulled_object,center)
+
+	return
 
+/obj/machinery/power/supermatter_shard/proc/supermatter_anomaly_gen(turf/anomalycenter, type = 1, anomalyrange = 5)
+	var/turf/L = pick(orange(anomalyrange, anomalycenter))
+	if(L)
+		if(type == 1)
+			var/obj/effect/anomaly/flux/A = new(L)
+			A.explosive = 0
+			A.lifespan = 300
+		else if(type == 2)
+			var/obj/effect/anomaly/grav/A = new(L)
+			A.lifespan = 250
+		else if(type == 3)
+			var/obj/effect/anomaly/pyro/A = new(L)
+			A.lifespan = 200
+
+	return
+
+/obj/machinery/power/supermatter_shard/proc/supermatter_zap(atom/src, range = 3, power)
+	. = src.dir
+	if(power < 1000)
+		return
+
+	var/target_atom
+	var/mob/living/target_mob
+	var/obj/machinery/target_machine
+	var/obj/structure/target_structure
+	var/list/arctargetsmob = list()
+	var/list/arctargetsmachine = list()
+	var/list/arctargetsstructure = list()
+
+	if(prob(20)) //let's not hit all the engineers with every beam and/or segment of the arc
+		for(var/mob/living/Z in oview(src, range+2))
+			arctargetsmob += Z
+	if(arctargetsmob.len)
+		var/mob/living/H = pick(arctargetsmob)
+		var/atom/A = H
+		target_mob = H
+		target_atom = A
+
+	else
+		for(var/obj/machinery/X in oview(src, range+2))
+			arctargetsmachine += X
+		if(arctargetsmachine.len)
+			var/obj/machinery/M = pick(arctargetsmachine)
+			var/atom/A = M
+			target_machine = M
+			target_atom = A
+
+		else
+			for(var/obj/structure/Y in oview(src, range+2))
+				arctargetsstructure += Y
+			if(arctargetsstructure.len)
+				var/obj/structure/O = pick(arctargetsstructure)
+				var/atom/A = O
+				target_structure = O
+				target_atom = A
+
+	if(target_atom)
+		src.Beam(target_atom, icon_state="nzcrentrs_power", time=5)
```

Comment:
uncessary src, in general we only specify the src when it actually conflicts with a local var or proc